lib/membrane_http_adaptive_stream/hls.ex

```diff
@@ -79,48 +75,47 @@ defmodule Membrane.HTTPAdaptiveStream.HLS do
   end
 
   defp build_media_playlist_path(%Manifest.Track{} = track) do
-    "#{track.header_name |> String.split(".") |> List.first()}.m3u8"
+    [track.content_type, "_", track.id_string, ".m3u8"] |> Enum.join("")
   end
 
   defp build_media_playlist_tag(%Manifest.Track{} = track) do
     case track do
       %Manifest.Track{content_type: :audio} ->
-        """
-        #EXT-X-MEDIA:TYPE=AUDIO,NAME=#{@default_audio_track_name},GROUP-ID=#{@default_audio_track_id},AUTOSELECT=YES,DEFAULT=YES,URI="audio.m3u8"
-        """
+        "#EXT-X-MEDIA:TYPE=AUDIO,NAME=\"#{@default_audio_track_name}\",GROUP-ID=\"#{@default_audio_track_id}\",AUTOSELECT=YES,DEFAULT=YES,URI=\"audio.m3u8\""
 
       %Manifest.Track{content_type: :video} ->
-        """
-        #EXT-X-STREAM-INF:BANDWIDTH=#{BandwidthCalculator.calculate_bandwidth(track)},CODECS="avc1.42e00a",
-        """
+        "#EXT-X-STREAM-INF:BANDWIDTH=#{BandwidthCalculator.calculate_bandwidth(track)},CODECS=\"avc1.42e00a\""
     end
   end
 
   defp build_master_playlist(audios, videos) do
     case {audios, videos} do
       {[audio], nil} ->
-        @master_playlist_header <> build_media_playlist_tag(audio)
+        [@master_playlist_header, build_media_playlist_tag(audio)] |> Enum.join("")
 
       {nil, videos} ->
-        @master_playlist_header <>
-          (videos
-           |> Enum.filter(&(&1.segments != @empty_segments))
-           |> Enum.map(&(build_media_playlist_tag(&1) <> build_media_playlist_path(&1)))
-           |> Enum.join("\n"))
+        [
+          @master_playlist_header
+          | videos
+            |> Enum.filter(&(&1.segments != @empty_segments))
+            |> Enum.flat_map(&[build_media_playlist_tag(&1), build_media_playlist_path(&1)])
+        ]
+        |> Enum.join("\n")
 
       {[audio], videos} ->
-        @master_playlist_header <>
-          build_media_playlist_tag(audio) <>
-          (videos
-           |> Enum.filter(&(&1.segments != @empty_segments))
-           |> Enum.map(
-             &[
-               "#{build_media_playlist_tag(&1)},AUDIO=#{@default_audio_track_id}",
-               build_media_playlist_path(&1)
-             ]
-           )
-           |> List.flatten()
-           |> Enum.join("\n"))
+        [
+          @master_playlist_header,
+          build_media_playlist_tag(audio)
+          | videos
+            |> Enum.filter(&(&1.segments != @empty_segments))
+            |> Enum.flat_map(
+              &[
+                "#{build_media_playlist_tag(&1)},AUDIO=\"#{@default_audio_track_id}\"",
+                build_media_playlist_path(&1)
+              ]
+            )
+        ]
+        |> Enum.join("\n")
```

Comment:
Did you consider the following form? I think you would show the general structure of the playlist a little better and it is less nested.
```suggestion
       video_tracks =
         videos
         |> Enum.filter(&(&1.segments != @empty_segments))
         |> Enum.flat_map(&[
             "#{build_media_playlist_tag(&1)},AUDIO=\"#{@default_audio_track_id}\"",
             build_media_playlist_path(&1)
           ])
           
        [
          @master_playlist_header,
          build_media_playlist_tag(audio),
          video_tracks
        ]
        |> List.flatten()
        |> Enum.join("\n")
```

I'm not sure if this is the best way, performance wise - probably not, but I think in general you shouldn't try so hard to stick to functional paradigm, not everything has to be one expression, which is usually unreadable